### ISSUE TITLE
Complementing windows don't resize together

### ISSUE THREAD

**Issue by TheRoarkster** [Leleat/Tiling-Assistant#96]
Great extension.  Is there a way to add the ability to automatically adjust the width of two windows that are touching, a la "[sticky window snapping](https://github.com/Flupp/sticky-window-snapping)?"  This comes in handy in wide screens, where two windows start at 50/50, but you want to drag them to 75/25.  Thanks.

**Comment by Leleat** [comment]
Can you elaborate please? Windows in a tile group already resize together (note that a tile group is determined when a window is tiled). Here is a gif as an example

![output](https://user-images.githubusercontent.com/53054301/133316100-fcb748e1-e31e-479d-921e-40a30c9d7451.gif)

**Comment by TheRoarkster** [comment]
Hmm.   The movement at the end of the gif you provided _does_ show the type of adjustment I mean.  But it doesn't seem to happen when I drag two adjacent windows.  Is there a setting to enable it (other than the default extension)?  Or a default GNOME behavior that may be overriding it?

**Comment by TheRoarkster** [comment]
Apologies.  It does seem to work.  At least on my desktop, the issue was that the adjacent windows weren't completely "touching" each other.  So the resizing icon that showed up was the default GNOME resize as opposed to the two-sided arrow that allows the auto-adjust.  I'll close this, and thanks for your help.

**Comment by Leleat** [comment]
> At least on my desktop, the issue was that the adjacent windows weren't completely "touching" each other. 

Hmm, how did you tile the 2 windows? Theoretically, they should automatically adapt their sizes to fill the available space (unless they are overlapped). Here is an example

![output](https://user-images.githubusercontent.com/53054301/133318497-21ca9f42-bbf3-47b8-b97d-383a653ac00f.gif)

If that isn't happening, then it is definitely a bug... and I need to fix that. Can you provide more info then?

> So the resizing icon that showed up was the default GNOME resize as opposed to the two-sided arrow that allows the auto-adjust

That isn't caused by my extension (that's app specific).

> sometimes I can get it to "grab" the two windows, and sometimes it's just one.

Are those 2 windows in a tileGroup? A tileGroup is set when a a window is tiled. All **non-overlapping** tiled windows at that moment will be considered in a tileGroup. Or are they just 2 "random" windows, where one is on the left and the other one on the right?

**Comment by TheRoarkster** [comment]
I'll make a video later, but the issue seems repeatable if I have a window half-tiled horizontally on the left side and a window quarter-tiled in the upper right quadrant.  Then, even though they are "touching" dragging the middle separator results in only one of the windows resizing.  However, if I tile both at half-tile, the correct behavior of both adjusting with the window resize -- as shown in your gif -- works.

**Comment by TheRoarkster** [comment]
Here is a video.  It won't grab if a window is quarter tiled (except it will grab another quarter tiled window below it).  


https://user-images.githubusercontent.com/35146738/133493448-8d158851-2235-41ba-8e9e-e8e9a5699865.mp4


Probably unrelated, but the Keybindings won't allow the use of the "Control" button.  So I can't set quarter tiling to "Ctrl+Left" for example.

**Comment by Leleat** [comment]
> Here is a video. It won't grab if a window is quarter tiled (except it will grab another quarter tiled window below it).

Unfortunately, you didn't recorder your mouse, so I am not sure what's going on exactly. What happens at 5s+? Did you to tile with the keyboard shortcut or what was all that movement?

**Comment by TheRoarkster** [comment]
Apologies.  I'll try again with a different recorder - that was the default GNOME version, and it does not appear to record the mouse.

**Comment by Leleat** [comment]
> Apologies. I'll try again with a different recorder - that was the default GNOME version, and it does not appear to record the mouse.

Any update here?

**Comment by TheRoarkster** [comment]
Apologies.  I will get this uploaded tomorrow.

**Comment by TheRoarkster** [comment]
I am both uploading the most recent video and noting that the problem is resolved.  I noticed there was an update to Tiling Assistant.  Not sure if that fixed it or it was something else.  But it works for me now.  

![Tiling](https://user-images.githubusercontent.com/35146738/135539692-5c0f527f-40a3-482b-a8ed-907fe266f2f7.gif)

**Comment by Leleat** [comment]
> I am both uploading the most recent video and noting that the problem is resolved.

I am seeing at least 1 bug in your gif. When you try to tile your graphics app to the bottom-left, the tile preview (that blue rectangle) is very small. That is definitely wrong. The tile preview should fill the available horizontal space until it meets another window like this:

![output](https://user-images.githubusercontent.com/53054301/135611880-ff2f49c6-3c98-4005-a279-0fdc3f741db9.gif)

So let's debug it. For that I need more info

1. What's your distro and the version?
2. What's your GNOME Shell version (run `gnome shell --version` in a terminal)
3. What's extension version do you use?
4. What other extensions do you have installed?
5. Are you on XOrg or Wayland?
6. Any specific apps you have problems with (what are the 2 apps you used in the video)?
7. Do you see any warnings / errors in when you run `journalctl -fo cat /usr/bin/gnome-shell` while using Tiling Assistant that seem related to this extension?

**Comment by TheRoarkster** [comment]
Here you go.   Note: the automatic resizing two windows feature has again started to sometimes not work with quarter-tiled windows.  

1. Manjaro Linux
DISTRIB_ID=ManjaroLinux
DISTRIB_RELEASE=21.1.5
DISTRIB_CODENAME=Pahvo

2.  GNOME Shell 40.5

3.  I don't know.  I can't find an extension version for Tiling Assistant.  

4.  Other extensions:  
AppIndicator and KStatusNotifierItem Support
Auto Move Windows
Clipboard Indicator
Coverflow Alt-Tab
Custom Hot Corners - Extended
Dash to Dock
Extension List
GSConnect 
Just Perfection 
Nothing to say 
OpenWeather 
system-monitor-next 
Transparent Shell
User themes

5.  Xorg

6.  GNOME Terminal 

7. No.  Just several iterations of this error: `JS ERROR: Error: GSettings key scroll-to-focused-application not found in schema org.gnome.shell.extensions.dash-to-dock`

I used the debug that you suggested and saw a line where the extension saved the original placement.  I'll attach another video to show you.

**Comment by TheRoarkster** [comment]
Below is the version.  Tried to create the video, but the software I used is not great and hides the top bar -- so I can't move windows easily.  Will find another one tonight.  

For the keyboard shortcut, I am not sure what is going on.  I can set "CTRL+LEFT/RIGHT" in GNOME Settings and in TA.  But when I try to set TA's "Corner Tiling" options, I get an error sound as soon as I hit any non-alphanumeric key (e.g., Shift, CTRL, Alt, Super, etc.) and it does not enter the key.  So I can only set Corner Tiling to, say, the letter "A" or the number "1."  


```
tiling-assistant@leleat-on-github
  Name: Tiling Assistant
  Description: An extension which adds a Windows-like snap assist to GNOME. It also expands GNOME's 2 column tiling design and adds more features. Check out the settings on how use this extension or on how to get help.
  Path: /home/rsync/.local/share/gnome-shell/extensions/tiling-assistant@leleat-on-github
  URL: https://github.com/Leleat/Tiling-Assistant
  Version: 25
  State: ENABLED

```

**Comment by Leleat** [comment]
>  I get an error sound as soon as I hit any non-alphanumeric key (e.g., Shift, CTRL, Alt, Super, etc.) and it does not enter the key. So I can only set Corner Tiling to, say, the letter "A" or the number "1."

I also get a error sound in a Manjaro vm but I can still set the shortcut. My guess is that you try to set it to a shortcut which is already taken in GNOME Settings.

> PS. Did you make sure that you didn't set the same keyboard shortcut in this extension and for GNOME's default tiling?

**Comment by Leleat** [comment]
> CTRL+SHIFT+LEFT

Yep, no problem here. Here is a gif where I bind `Ctrl` + `Shift` + `Right` / `Left` to tile to the top-right.

![output](https://user-images.githubusercontent.com/53054301/137634431-ba7ca8d1-12cb-4458-a4aa-b7e9a5cf4e85.gif)

**Comment by TheRoarkster** [comment]
Weird.  If I try to use "SHIFT" at all, the keyboard command is simply not created.  Trying to create video now.

**Comment by TheRoarkster** [comment]
And here is one where I try to use the tiling display feature.

https://user-images.githubusercontent.com/35146738/137636024-178b3f27-e684-48d1-85d9-db7a480ed974.mp4

**Comment by Leleat** [comment]
> Ok. Here is a general video.

It's working as it should. Does it sometimes fail or something?
